test/cloudant.test.js

```diff
@@ -295,6 +310,124 @@ describe('cloudant connector', function() {
       });
     });
   });
+
+  describe('allow numerical `id` value', function() {
+    var data = [{
+      id: 1,
+      name: 'John Chow',
+      age: 45,
+    }, {
+      id: 5,
+      name: 'Kelly Johnson',
+      age: 25,
+    }, {
+      id: 12,
+      name: 'Michael Santer',
+      age: 30,
+    }];
+
+    before(function(done) {
+      SimpleEmployee.create(data, function(err, result) {
+        if (err) done(err);
+        done(null, result);
+      });
+    });
+
+    it('find instances with numeric id (findById)', function(done) {
+      SimpleEmployee.findById(data[1].id, function(err, result) {
+        should.not.exist(err);
+        should.deepEqual(result.__data, data[1]);
+        done();
+      });
+    });
+
+    it('find instances with "where" filter', function(done) {
+      SimpleEmployee.find({where: {id: data[0].id}}, function(err, result) {
+        should.not.exist(err);
+        should.equal(result.length, 1);
+        should.deepEqual(result[0].__data, data[0]);
+        done();
+      });
+    });
+
+    it('find instances with "order" filter (ASC)', function(done) {
+      SimpleEmployee.find({order: 'id ASC'}, function(err, result) {
+        should.not.exist(err);
+        should(result[0].id).equal(data[0].id);
```

Comment:
better to add a `result.length.should.equal(3)` before checking `result` details. Not necessary, but think of if result is null or missing data, that check tells you what's going wrong faster than reading 3 failed assertions and figure out the reason :p